src/Agent/NewRelic/Agent/Core/Metrics/MetricNameService.cs

```diff
@@ -138,8 +121,11 @@ private static string RenameUsingRegexRules(string input, IEnumerable<RegexRule>
                     continue;
 
                 if (rule.Ignore)
-                    throw new IgnoreTransactionException($"Ignoring \"{input}\" because it matched pattern \"{rule.MatchExpression}\"", input);
-
+                {
+                    Log.Debug($"Ignoring \"{input}\" because it matched pattern \"{rule.MatchExpression}\"");
+                    return null;
+                }
+                    
                 if (ruleResult.Replacement == null)
                     throw new Exception("RuleResult matched but returned null replacement string");
```

Comment:
Even though the previous code let this exception bubble up, and it doesn't seem like this error is likely to occur, it would be safer to not throw this exception here, and instead log the error using something like the following.
```
Log.Error("RuleResult matched but returned null replacement string");
return null;
```